src/schedule/dma_multi_chan_domain.c

```diff
@@ -275,38 +275,23 @@ static void dma_multi_chan_domain_unregister(struct ll_schedule_domain *domain,
  * \return True is task should be executed, false otherwise.
  */
 static bool dma_multi_chan_domain_is_pending(struct ll_schedule_domain *domain,
-					     struct task *task, struct comp_dev **comp)
+					     struct task *task)
 {
 	struct dma_domain *dma_domain = ll_sch_domain_get_pdata(domain);
 	struct pipeline_task *pipe_task = pipeline_task_get(task);
 	struct dma *dmas = dma_domain->dma_array;
-	uint32_t status;
 	int i;
 	int j;
 
+	/* it's too soon for this task */
+	if (!pipe_task->registrable &&
+	    pipe_task->task.start > platform_timer_get_noirq(timer_get()))
+		return false;
+
 	for (i = 0; i < dma_domain->num_dma; ++i) {
 		for (j = 0; j < dmas[i].plat_data.channels; ++j) {
-			if (!*comp) {
```

Comment:
I see you removed the code that fixed #3170.
See #3339
@lyakh please comment

Reply:
> I see you removed the code that fixed #3170.
> See #3339
> @lyakh please comment

Thanks for reminding, maybe we need to move this interrupt clearing to the domain_clear(). Let me try that.

Reply:
it isn't about clearing interrupts, it's about sending notifications, i.e. about running tasks. Why do you think this fix isn't needed any more?

Reply:
> it isn't about clearing interrupts, it's about sending notifications, i.e. about running tasks. Why do you think this fix isn't needed any more?

Hi @lyakh the logic here was kind of convoluted, so basically we should notify all DMA channels which run for the task that has the same sched_comp with the one we are getting interrupt here, is the passed in *task* here important? what we need to do for this *task* is to mark it as pending or not?